fiona/_shim1.pyx

```diff
@@ -1,25 +1,41 @@
+import logging
+
 from fiona.ogrext1 cimport *
 
 from fiona.errors import DriverIOError
 from fiona._err cimport exc_wrap_pointer
 
+
+log = logging.getLogger(__name__)
+
 cdef int OGRERR_NONE = 0
 
+
 cdef bint is_field_null(void *feature, int n):
     if not OGR_F_IsFieldSet(feature, n):
         return True
     else:
         return False
 
+
 cdef void gdal_flush_cache(void *cogr_ds):
     retval = OGR_DS_SyncToDisk(cogr_ds)
     if retval != OGRERR_NONE:
         raise RuntimeError("Failed to sync to disk")
 
-cdef void* gdal_open_vector(const char *path_c, int mode, drivers):
+
+cdef void* gdal_open_vector(const char *path_c, int mode, drivers, options):
```

Comment:
I'm extending two shim functions with a new positional argument: a Python dict of the creation or open options.